Find a functionally equivalent code:com.android.layoutlib.bridge.util.SparseWeakArray.append:COMMENT
Method Modifier: public      
Comment:/**
 * Puts a key/value pair into the array, optimizing for the case where
 * the key is greater than all existing keys in the array.
 */

@@ -6,23 +6,12 @@
     if (mSize >= mKeys.length && (mGarbage || hasReclaimedRefs())) {
         gc();
     }
-    int pos = mSize;
-    if (pos >= mKeys.length) {
-        int n = ArrayUtils.idealIntArraySize(pos + 1);
-        int[] nkeys = new int[n];
-        WeakReference<?>[] nvalues = new WeakReference[n];
-        // Log.e("SparseArray", "grow " + mKeys.length + " to " + n);
-        System.arraycopy(mKeys, 0, nkeys, 0, mKeys.length);
-        System.arraycopy(mValues, 0, nvalues, 0, mValues.length);
-        mKeys = nkeys;
-        mValues = nvalues;
-    }
-    mKeys[pos] = key;
-    mValues[pos] = new WeakReference(value);
-    mSize = pos + 1;
+    mKeys = GrowingArrayUtils.append(mKeys, mSize, key);
+    mValues = GrowingArrayUtils.append(mValues, mSize, new WeakReference(value));
+    mSize++;
 }

